iterator.go

```diff
@@ -307,17 +245,24 @@ func (it *changeIterator[Obj]) Next(txn ReadTxn) (seq iter.Seq2[Change[Obj], Rev
 		if it.iter == nil {
 			return
 		}
-		for obj, rev, deleted, ok := it.iter.Next(); ok; obj, rev, deleted, ok = it.iter.Next() {
-			if deleted {
-				it.deleteRevision = rev
-				it.dt.mark(rev)
-			} else {
-				it.revision = rev
-			}
+		for _, obj, ok := it.iter.Next(); ok; _, obj, ok = it.iter.Next() {
+			rev := obj.revision
+			it.revision = rev
 			change := Change[Obj]{
-				Object:   obj,
 				Revision: rev,
-				Deleted:  deleted,
+			}
+			if obj.deleted {
+				if rev <= it.startRevision {
+					// Ignore objects that were marked deleted before this
+					// changge iterator was created.
```

Comment:
changge typo

It's somehow asymmetric to me that the iterator has a startRevision, but only cares about it for deleted objects. It seems correct, but irks me for some reason

Reply:
Yeah let me rename that!